### ISSUE TITLE
Parachutes not working - NULL Reference Exceptions on deploy 

### ISSUE THREAD

**Issue by TheDarkBadger** [PorktoberRevolution/ReStocked#751]
When any parachute deploys, the sound effect plays but no parachute appears and the craft does not slow down. 
The debug log spams Null Reference Exceptions until the parachute part is destroyed on impact
Issue does not happen on removal of Restock
Game version is 1.8.1 and the newest version of Restock

**Comment by ChrisAdderley** [comment]
Are you using any other mods? Adding a logfile of the event will help debug things.

**Comment by catonthekbd** [comment]
modlist (text) https://www.dropbox.com/s/1qywikedc0e4wu5/modlist.txt?dl=0
modlist (ckan) https://www.dropbox.com/s/18qks9u5shup6vj/modlist.ckan?dl=0
image without restock https://www.dropbox.com/s/qo688xyjt4cubah/screenshot_2019-11-07--17-33-16.png?dl=0
image with restock https://www.dropbox.com/s/tgg0womkkqaijjr/screenshot_2019-11-07--17-50-07.png?dl=0

**Comment by drewcassidy** [comment]
Do you have a log file? can you reproduce the error with only Restock installed?

**Comment by catonthekbd** [comment]
with only restock/restock+/modul manager it works
I don't get a real-cute info window at all
where did it come from?

**Comment by catonthekbd** [comment]
and uninstalling F.A.R. get things working again
is everyone else with the problem using F.A.R as well?

**Comment by ChrisAdderley** [comment]
So the issues is FAR without RealChute? I'll take a look at the FAR repo...

**Comment by catonthekbd** [comment]
I have not tested FAR without Restock yet - this will have to wait 24+ hours

**Comment by ChrisAdderley** [comment]
Sorry, I mean
1. RS = Ok
2. RS + FAR = bad
3. RS + RC + FAR = Ok

**Comment by ChrisAdderley** [comment]
Found the issue: FAR ships a "real chute lite" plugin. This is disabled if RealChute exists and we correctly handle RealChute. We don't handle this 'lite' version. 

https://github.com/dkavolis/Ferram-Aerospace-Research/blob/master/GameData/FerramAerospaceResearch/RealChuteLite/RealChuteLite.cfg

**Comment by ChrisAdderley** [comment]
I believe the thing I did will fix it but I don't have a testing opportunity. If you'd like to test resolution, please get this file: 

https://github.com/PorktoberRevolution/ReStocked/blob/master/Distribution/Restock/GameData/ReStock/Compatibility/FAR.cfg

Drop it in the right place (ReStock/Compatibility). I believe it should solve the problem.

**Comment by catonthekbd** [comment]
will do tommorow night - about to head to bed - thanks for the quick response

**Comment by dkavolis** [comment]
FYI, for some reason it's `canopyName` and not `parachuteName`, don't ask why.
[RealChuteFAR.cs](https://github.com/dkavolis/Ferram-Aerospace-Research/blob/2cf406c93ac84899a2da9e667f96be05c3ea5e40/FerramAerospaceResearch/RealChuteLite/RealChuteFAR.cs#L99)

**Comment by catonthekbd** [comment]
I set up a new clone with minimum mods https://www.dropbox.com/s/az0gk59fza1914f/modlist2.txt?dl=0
using your FAR.cfg did not make any difference
s/parachuteName/canopyName/g did not help either
my current FAR.cfg -  https://www.dropbox.com/s/fcsccvp6wny6spy/FAR.cfg?dl=0

**Comment by catonthekbd** [comment]
I have a log file if you want it. in full, or do I need to edit it down?

[LOG 19:47:04.317] Deleting root node in file FerramAerospaceResearch/RealChuteLite/RealChuteLite node: @PART[*]:HAS[@MODULE[ModuleParachute]]:NEEDS[RealChute&!RealismOverhaul]:AFTER[RealChute] as it can't satisfy its NEEDS
[LOG 19:47:04.317] Deleting root node in file ReStock/Compatibility/RealChute node: @PART[parachuteDrogue]:AFTER[RealChute] as it can't satisfy its AFTER
[LOG 19:47:04.317] Deleting root node in file ReStock/Compatibility/RealChute node: @PART[parachuteLarge]:AFTER[RealChute] as it can't satisfy its AFTER
[LOG 19:47:04.317] Deleting root node in file ReStock/Compatibility/RealChute node: @PART[parachuteSingle]:AFTER[RealChute] as it can't satisfy its AFTER
[LOG 19:47:04.317] Deleting root node in file ReStock/Compatibility/RealChute node: @PART[parachuteRadial]:AFTER[RealChute] as it can't satisfy its AFTER
[LOG 19:47:04.317] Deleting root node in file ReStock/Compatibility/RealChute node: @PART[radialDrogue]:AFTER[RealChute] as it can't satisfy its AFTER

[ERR 19:52:32.055] Module RealChuteFAR threw during OnStart: System.NullReferenceException: Object reference not set to an instance of an object
  at FerramAerospaceResearch.RealChuteLite.RealChuteFAR.InitializeAnimationSystem () [0x00045] in <6338ca968c7c4c8a971a01d858e759df>:0 
  at FerramAerospaceResearch.RealChuteLite.RealChuteFAR.OnStart (PartModule+StartState startState) [0x00091] in <6338ca968c7c4c8a971a01d858e759df>:0 
  at Part.ModulesOnStart () [0x00120] in <9d71e4043e394d78a6cf9193ad011698>:0 



[LOG 19:54:59.541] [RealChute]: parachuteSingle was activated in stage 0
[EXC 19:54:59.638] NullReferenceException: Object reference not set to an instance of an object
	FerramAerospaceResearch.RealChuteLite.RealChuteFAR.PreDeploy () (at <6338ca968c7c4c8a971a01d858e759df>:0)
	FerramAerospaceResearch.RealChuteLite.RealChuteFAR.FixedUpdate () (at <6338ca968c7c4c8a971a01d858e759df>:0)

**Comment by dkavolis** [comment]
I'm pretty sure [this](https://github.com/dkavolis/Ferram-Aerospace-Research/blob/2cf406c93ac84899a2da9e667f96be05c3ea5e40/FerramAerospaceResearch/RealChuteLite/RealChuteFAR.cs#L871) line throws first. MM cache would be better in this case.

**Comment by catonthekbd** [comment]
all my MM files should be in here https://www.dropbox.com/sh/z0lnrq3x94kmcdo/AAAaXum7nXzameLb7d9bVYJla?dl=0

**Comment by dkavolis** [comment]
Try
```
// FAR ships a RealChute Lite that is similar to the RC patch

@PART[parachuteDrogue]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @preDeploymentAnimation = semiDeployLarge
      @deploymentAnimation = fullyDeployLarge
      @capName = ChuteCap
      @canopyName  = B_ParachuteLargeDrogueRotator
  }

  !DRAG_CUBE {}
}

@PART[parachuteLarge]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @preDeploymentAnimation = semiDeployLarge
      @deploymentAnimation = fullyDeployLarge
      @capName = ChuteCap
      @canopyName  = B_ParachuteLargeRotator
  }

  !DRAG_CUBE {}
}

@PART[parachuteSingle]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @preDeploymentAnimation = semiDeployLarge
      @deploymentAnimation = fullyDeployLarge
      @capName = SmallChuteCap
      @canopyName  = B_ParachuteRoot004
  }

  !DRAG_CUBE {}
}

@PART[parachuteRadial]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @preDeploymentAnimation = semiDeployLarge
      @deploymentAnimation = fullyDeployLarge
      @capName = RadialChuteCap
      @canopyName  = B_ParachuteRoot
  }

  !DRAG_CUBE {}
}

@PART[radialDrogue]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @preDeploymentAnimation = semiDeployLarge
      @deploymentAnimation = fullyDeployLarge
      @capName = RadialChuteCapDrogue
      @canopyName  = B_ParachuteRoot005
  }

  !DRAG_CUBE {}
}
```
There should be no `PARACHUTE` subnode.

**Comment by catonthekbd** [comment]
Much better :)
chute info window now works
chute pre-deploy and deploy work (from physics POV)
chute animation seems to show a chute made for the Boing Starliner - it never comes out of the container
https://www.dropbox.com/s/a619lx2qt5f5b7a/boing-chute.png?dl=0

**Comment by dkavolis** [comment]
Must be an issue with animation name, try replacing `preDeploymentAnimation` with `semiDeployedAnimation` and `deploymentAnimation` with `fullyDeployedAnimation`

**Comment by catonthekbd** [comment]
that works :)
many thanks for all the help
my reference copy of FAR.cfg attached

// FAR ships a RealChute Lite that is similar to the RC patch

@PART[parachuteDrogue]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @semiDeployedAnimation = semiDeployLarge
      @fullyDeployedAnimation = fullyDeployLarge
      @capName = ChuteCap
      @canopyName  = B_ParachuteLargeDrogueRotator
  }

  !DRAG_CUBE {}
}

@PART[parachuteLarge]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @semiDeployedAnimation = semiDeployLarge
      @fullyDeployedAnimation = fullyDeployLarge
      @capName = ChuteCap
      @canopyName  = B_ParachuteLargeRotator
  }

  !DRAG_CUBE {}
}

@PART[parachuteSingle]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @semiDeployedAnimation = semiDeployLarge
      @fullyDeployedAnimation = fullyDeployLarge
      @capName = SmallChuteCap
      @canopyName  = B_ParachuteRoot004
  }

  !DRAG_CUBE {}
}

@PART[parachuteRadial]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @semiDeployedAnimation = semiDeployLarge
      @fullyDeployedAnimation = fullyDeployLarge
      @capName = RadialChuteCap
      @canopyName  = B_ParachuteRoot
  }

  !DRAG_CUBE {}
}

@PART[radialDrogue]:AFTER[FerramAerospaceResearch]
{
  @MODULE[RealChuteFAR]
  {
      @semiDeployedAnimation = semiDeployLarge
      @fullyDeployedAnimation = fullyDeployLarge
      @capName = RadialChuteCapDrogue
      @canopyName  = B_ParachuteRoot005
  }

  !DRAG_CUBE {}
}

**Comment by ChrisAdderley** [comment]
Implemented this in 63fb192045892cc64d42d629e5fef1fd61c31cf0
